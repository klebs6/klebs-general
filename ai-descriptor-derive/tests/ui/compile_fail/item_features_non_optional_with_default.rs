// ---------------- [ File: ai-descriptor-derive/tests/ui/compile_fail/item_features_non_optional_with_default.rs ]
#![allow(unused_imports)]
use ai_descriptor_derive::*;
use ai_descriptor_trait::*;
use std::borrow::Cow;

#[derive(ItemWithFeatures)]
#[ai("Potion")]
struct Potion {
    #[ai(feature_if_none = "This should not be used and should cause an error.")]
    effect: PotionEffect,
}

struct PotionEffect;

impl ItemFeature for PotionEffect {
    fn text(&self) -> Cow<'_, str> {
        Cow::Borrowed("Gives you super strength.")
    }
}

fn main() {}
