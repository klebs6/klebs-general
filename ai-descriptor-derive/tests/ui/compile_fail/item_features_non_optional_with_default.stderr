error: The `feature_if_none` attribute is only applicable to Option types
  --> tests/ui/compile_fail/item_features_non_optional_with_default.rs:10:5
   |
10 | /     #[ai(feature_if_none = "This should not be used and should cause an error.")]
11 | |     effect: PotionEffect,
   | |________________________^
